### PR TITLE
chore: explicitly opt-in to node test events

### PR DESCRIPTION
**Pull request by galargh**
- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [ ] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [x] I didn't do anything of this.

---

This is a follow-up to the discussion that emerged under https://nomicfoundation.slack.com/archives/C03P6B72ZHU/p1729529455497859

### CODE REVIEW

File: v-next/hardhat-node-test-reporter/src/reporter.ts [hunk 1]
```diff
@@ -244,45 +244,15 @@ export function hardhatTestReporter(
...
-        // but @types/node doesn't know about it yet. Once it does, the dependency should be updated
-        // and this comment removed.
-        case "test:summary": {
-          // Do nothing
-          break;
-        }
-        /* eslint-disable-next-line @typescript-eslint/switch-exhaustiveness-check --
-        We have this extra check here becase we know the @types/node type is
-        unreliable */
         default: {
```
Comment (kanej) [inline v-next/hardhat-node-test-reporter/src/reporter.ts:251]: This looks good.
I was tempted to say we should keep the `do nothing` tests so we can explicitly see what events we are skipping (and consider whether we should add the actions for them).
But I think the complexity cost of keeping `@ts-expect-error` annoys me more than a theoretical future readability payoff.
What do you think?

Comment (galargh) [reply v-next/hardhat-node-test-reporter/src/reporter.ts:251]: That was my thinking exactly. Also, when we want to reconsider what events we use, we should look at node directly anyway since @types/node, sadly, might be lagging behind. Considering all that, a more concise switch statement might be a better choice here.

Comment (kanej) [reply v-next/hardhat-node-test-reporter/src/reporter.ts:251]: Agreed.

**Review by kanej — APPROVED** [review]

File: v-next/hardhat-node-test-reporter/src/reporter.ts [hunk 2]
```diff
@@ -244,45 +244,15 @@ export function hardhatTestReporter(
...
-          const _isNever: never = event;
-          void _isNever;
-
-          yield chalk.red(`Unsuported node:test event:`, event);
+          // NOTE: Do nothing; we explicitly opt-in to the events we want to
```
Comment (alcuadrado) [inline v-next/hardhat-node-test-reporter/src/reporter.ts:252]: Maybe we can log it with `debug`

Comment (galargh) [reply v-next/hardhat-node-test-reporter/src/reporter.ts:252]: I think if we do that we might run into some ordering edge cases as the reporter doesn't really log anything at the moment; it's just a string generator that is later piped through to the stdout.

I do agree with the sentiment though, it could be nice to be able to enable extra logging of unknown events.

Comment (alcuadrado) [reply v-next/hardhat-node-test-reporter/src/reporter.ts:252]: Good point. Let's play safe.
